### PR TITLE
animefillerlist presence add by ! Yelo

### PR DESCRIPTION
**Pull request by u0f**

### CODE REVIEW

File: websites/A/AnimeFillerList/presence.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/**
+ * Coded by ! Yelo (425326889565945856)
+ * Presence for Premid
+ * Website: www.animefillerlist.com
+ */
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:5]: remove
```suggestion
```

File: websites/A/AnimeFillerList/presence.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    clientId: "894342965772820490"
+  }),
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "animefillerlistlogo"
+  };
+
+  if (document.location.hostname === "www.animefillerlist.com") {
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:17]: not needed if

File: websites/A/AnimeFillerList/presence.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    } else if (document.location.pathname.includes("/changelog")) {
+      presenceData.startTimestamp = browsingStamp;
+      presenceData.details = "Viewing Changelog...";
+    } else if (document.location.pathname.includes("/privacy-policy")) {
+      presenceData.startTimestamp = browsingStamp;
+      presenceData.details = "Viewing Privacy Policy...";
+    }
+  }
+
+  if (presenceData.details === null) {
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:74]:
```suggestion
  if (!presenceData.details) {
```

File: websites/A/AnimeFillerList/presence.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "animefillerlistlogo"
+  };
+
+  if (document.location.hostname === "www.animefillerlist.com") {
+    if (document.location.pathname === "/") {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:19]: define once, not every if

File: websites/A/AnimeFillerList/presence.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      presenceData.details = "In home page...";
+    } else if (document.location.pathname.includes("/shows/latest-updates")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:22]: ditto

File: websites/A/AnimeFillerList/presence.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      presenceData.details = "Viewing Lastest updates...";
+    } else if (document.location.pathname.includes("/shows/")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:25]: ditto

File: websites/A/AnimeFillerList/presence.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      const select = document.querySelector(".Right > h1"),
+        selectClean = select.textContent.replace("Filler List", " ");
+      presenceData.details = "Viewing filler list:";
+      presenceData.state = selectClean;
+    } else if (document.location.pathname.includes("/shows")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:31]: ditto

File: websites/A/AnimeFillerList/presence.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+      presenceData.details = "Browsing for anime filler list...";
+    } else if (document.location.pathname.includes("/search/node/")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:34]: ditto

File: websites/A/AnimeFillerList/presence.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+      const search = document.location.pathname,
+        splitPath = search.split("/"),
+        selectPath = splitPath[splitPath.length - 1],
+        clean = selectPath.replace(/%20/g, " ");
+      presenceData.details = "Searching a filler list:";
+      presenceData.state = clean;
+    } else if (document.location.pathname.includes("/user/password")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:42]: ditto

File: websites/A/AnimeFillerList/presence.ts [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+      presenceData.details = "Requesting a new password...";
+    } else if (document.location.pathname.includes("/user/register")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:45]: ditto

File: websites/A/AnimeFillerList/presence.ts [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+      presenceData.details = "Creating a account...";
+    } else if (document.location.pathname.includes("/user/login")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:48]: ditto

File: websites/A/AnimeFillerList/presence.ts [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+      presenceData.details = "Logging in...";
+    } else if (document.location.pathname.includes("/users/")) {
+      const user = document.querySelector(".content > h1");
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:52]: ditto

File: websites/A/AnimeFillerList/presence.ts [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+      presenceData.state = user.textContent;
+      presenceData.buttons = [
+        {
+          label: "Look user",
+          url: `https://${document.location.hostname}/users/${user.textContent}`
+        }
+      ];
+
+    } else if (document.location.pathname.includes("/contact")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:63]: ditto

File: websites/A/AnimeFillerList/presence.ts [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+      presenceData.details = "Contacting with the page...";
+    } else if (document.location.pathname.includes("/changelog")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:66]: ditto

File: websites/A/AnimeFillerList/presence.ts [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+      presenceData.details = "Viewing Changelog...";
+    } else if (document.location.pathname.includes("/privacy-policy")) {
+      presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/A/AnimeFillerList/presence.ts:69]: ditto

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/A/AnimeFillerList/presence.ts (hunk 2 above)
Comment (u0f) [reply websites/A/AnimeFillerList/presence.ts:12]: Fixed

**Comment by u0f** [comment]
12 minor errores, but everything is okay.

**Comment by EncryptedDev** [comment]
> 12 minor errores, but everything is okay.

You need to the majority of them

**Comment by u0f** [comment]
What can I do now?

**Comment by u0f** [comment]
Yarn preatty executed.

**Comment by EncryptedDev** [comment]
Run `eslint --fix` in your presence folder to fix issues

**Pull request closed without merging**